source/create.js

```diff
@@ -27,16 +26,23 @@ const create = defaults => {
 		defaults.handler = (options, next) => next(options);
 	}
 
+	if (defaults.handler.constructor.name !== 'AsyncPromise') {
+		defaults.handler = util.promisify(defaults.handler);
+	}
+
 	function got(url, options) {
-		try {
-			return defaults.handler(normalizeArguments(url, options, defaults), getPromiseOrStream);
-		} catch (error) {
-			if (options && options.stream) {
-				throw error;
-			} else {
-				return Promise.reject(error);
-			}
-		}
+		const promiseOrStream = options && options.stream ? asStream() : asPromise();
+
+		// eslint-disable-next-line promise/prefer-await-to-then
+		normalizeArguments(url, options, defaults).then(normalizedOptions => {
+			defaults.handler(normalizedOptions, promiseOrStream._sendRequest).catch(error => {
+				promiseOrStream._reject(error);
+			});
+		}).catch(error => {
+			promiseOrStream._reject(error);
+		});
```

Comment:
This could be simplified to:

```js
(async () => {
	try {
		const normalizedOptions = await normalizeArguments(url, options, defaults);
		await defaults.handler(normalizedOptions, promiseOrStream._sendRequest);
	} catch (error) {
		promiseOrStream._reject(error);
	}
})();
```